pkg/controllers/manifestwork.go

```diff
@@ -0,0 +1,251 @@
+/*
+Copyright 2022.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controllers
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	hyp "github.com/openshift/hypershift/api/v1alpha1"
+	hypdeployment "github.com/stolostron/hypershift-deployment-controller/api/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	workv1 "open-cluster-management.io/api/work/v1"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	ManifestTargetNamespace       = "manifestwork-target-namespace"
+	CreatedByHypershiftDeployment = "created-by-hypershiftdeployment"
```

Comment:
Doesn't need to be in this pull request, but I'd like to use a key like. hypershift.open-cluster-management.io/created-by
Or an ownerRef if that's accessible in the Watch and Enqueue With